test/hummingbot/connector/exchange/bitmart/test_bitmart_api_order_book_data_source.py

```diff
@@ -296,31 +309,25 @@ def test_listen_for_order_book_diffs_successful(self, mock_ws):
                 }
             ]
         }
-        self.mocking_assistant.add_websocket_text_message(
-            mock_ws.return_value, ujson.dumps(resp)
+
+        self.mocking_assistant.add_websocket_aiohttp_message(
+            mock_ws.return_value, bitmart_utils.compress_ws_message(ujson.dumps(resp)), message_type=aiohttp.WSMsgType.BINARY
         )
+
         BitmartAPIOrderBookDataSource._trading_pairs = ["ETH-USDT"]
         self.listening_task = self.ev_loop.create_task(
             self.data_source.listen_for_order_book_diffs(self.ev_loop, msg_queue)
         )
-        first_msg: OrderBookMessage = self.ev_loop.run_until_complete(msg_queue.get())
-        self.assertTrue(first_msg.type == OrderBookMessageType.SNAPSHOT)
 
-    @patch("websockets.connect", new_callable=AsyncMock)
-    def test_websocket_connection_creation_raises_cancel_exception(self, mock_ws):
-        mock_ws.side_effect = asyncio.CancelledError
+        try:
+            self.async_run_with_timeout(msg_queue.get())
```

Comment:
Not sure I understand the use of this one. The goal of `async_run_with_timeout` is to cause the test case to fail on timeout (i.e. taking too long is a failure). In this case, we can use a non-mocked `asyncio.Queue` and do
```
first_msg = self.async_run_with_timeout(msg_queue.get())
```